src/main/pages/che-7/end-user-guide/proc_starting-a-workspace-with-a-devfile.adoc

```diff
@@ -1,27 +1,63 @@
 [id="starting-a-workspace-with-a-devfile_{context}"]
-= Starting a workspace with a Devfile
+= Starting a workspace with a devfile
+
+This section describes how to start a Che workspace using your own existing devfile.
 
-This paragraph is the procedure module introduction: a short description of the procedure.
 
-[discrete]
 .Prerequisites
+ 
+* A running instance of Eclipse Che.
+* An existing devfile.
 
-* A bulleted list of conditions that must be satisfied before the user starts following this assembly.
-* You can also link to other modules or assemblies the user must follow before starting this assembly.
-* Delete the section title and bullets if the assembly has no prerequisites.
 
-[discrete]
 .Procedure
 
-. Start each step with an active verb.
+There are several ways to launch a Che workspace based on a prepared devfile:
+
+.Devfile from Git
+
+Deploy the devfile to a Git source repository and then create a factory from this repository.
+
+. Put a `devfile.yaml` file in the root of your Git repository.
+. Execute the factory by opening the following URL:
++
+[subs="+quotes"]
+----
+https://__<your-che-host>__/f?url=https://github.com/__<mygroup>__/__<myrepo>__
+----
+
+.Devfile from URL
+
+Execute a devfile by constructing the factory with the URL pointing to the raw content of the devfile, for example:
+
+[subs="+quotes"]
+----
+https://__<your-che-host>__/f?url=https://pastebin.com/raw/ux6iCGaW
+----
+
+.Devfile over the Che REST API
```

Comment:
I think we can remove the REST API section. All the use cases should be covered by factory + chectl.

Reply:
ok